front/app/components/HelmetIntl/index.tsx

```diff
@@ -2,31 +2,39 @@ import React from 'react';
 import { Helmet } from 'react-helmet';
 import { WrappedComponentProps, MessageDescriptor } from 'react-intl';
 import { injectIntl } from 'utils/cl-intl';
+import useLocalize from 'hooks/useLocalize';
+import useAppConfiguration from 'hooks/useAppConfiguration';
+import { isNilOrError } from 'utils/helperUtils';
 
-type Props = {
+interface Props {
   title: MessageDescriptor;
   description: MessageDescriptor;
-};
-
-interface State {}
+}
 
-export class HelmetIntl extends React.PureComponent<
-  Props & WrappedComponentProps,
-  State
-> {
-  render() {
-    const { formatMessage } = this.props.intl;
-    const { title, description } = this.props;
+const HelmetIntl = ({
+  intl: { formatMessage },
+  title,
+  description,
+}: Props & WrappedComponentProps) => {
+  const localize = useLocalize();
+  const appConfig = useAppConfiguration();
 
+  if (!isNilOrError(appConfig)) {
     return (
       <>
         <Helmet
-          title={formatMessage(title)}
+          title={formatMessage(title, {
+            orgName: localize(
+              appConfig.attributes.settings.core.organization_name
+            ),
+          })}
           meta={[{ name: 'description', content: formatMessage(description) }]}
         />
       </>
     );
   }
-}
 
-export default injectIntl<Props>(HelmetIntl);
+  return null;
+};
+
+export default injectIntl(HelmetIntl);
```

Comment:
Do we still need injectIntl here?

Reply:
Yes/no. The component uses formatMessage, but it could be useIntl anywhere from now. What makes you think we don't need it anymore? :-)

Reply:
Oh, I just meant replacing it with useIntl :+1: Should we be trying to remove these HOCs? Or is it ok to leave them in existing code for now?

Reply:
Yeah, similar like we refactor other parts (withRouter, injectLocalize, class to functional). It'll be a gradual effort.